scripts/routes.js

```diff
@@ -0,0 +1,110 @@
+import {
+    textSummary
+} from 'https://jslib.k6.io/k6-summary/0.0.3/index.js'
+import http from 'k6/http'
+import {
+    check,
+    sleep,
+    randomSeed
+} from 'k6'
+import {
+    generateRandomCoordinate,
+    destinations,
+    oneLocationScenario as scenarios,
+    setThresholdsForScenarios,
+    summaryTrendStats,
+    deleteTimeFilterMetrics,
+    reportPerDestination,
+    getCountryCoordinates
+} from './common.js'
+
+export const options = {
+    scenarios,
+    summaryTrendStats,
+
+    thresholds: {
+        // Intentionally empty. I'll define bogus thresholds (to generate the sub-metrics) below.
+    }
+}
+
+setThresholdsForScenarios(options)
+randomSeed(__ENV.SEED || 1234567)
+
+export default function () {
+    const appId = __ENV.APP_ID
+    const apiKey = __ENV.API_KEY
+    const host = __ENV.HOST || 'api.traveltimeapp.com'
+    const countryCode = __ENV.COUNTRY || 'gb'
+    const countryCoords = getCountryCoordinates(countryCode, __ENV.COORDINATES)
+    const url = `https://${host}/v4/routes`
+    const transportation = __ENV.TRANSPORTATION || 'driving+ferry'
+    const dateTime = new Date().toISOString()
+
+    const params = {
+        headers: {
+            'Content-Type': 'application/json',
+            'X-Application-Id': appId,
+            'X-Api-Key': apiKey
+        }
+    }
+
+    const response = http.post(url, generateBody(transportation, countryCoords, dateTime), params)
+    check(response, {
+        'status is 200': (r) => r.status === 200,
+        'response body is not empty': (r) => r.body.length > 0
+    })
+    sleep(1)
+}
+
+export function handleSummary(data) {
+    // removing default metrics
+    deleteTimeFilterMetrics(data)
+
+    data = destinations.reduce((curData, curDestinations) => {
+        return reportPerDestination(curData, curDestinations)
+    }, data)
+
+    return {
+        stdout: textSummary(data, {
+            indent: ' ',
+            enableColors: true
+        })
+    }
+}
+
+function generateBody(
+    transportation,
+    countryCoords,
+    dateTime
+) {
+    const coordinates = countryCoords
+    const diff = 0.01
+
+    const origin = {
+        id: 'origin',
+        coords: generateRandomCoordinate(coordinates.lat, coordinates.lng, diff)
```

Comment:
something's messed up here, got runtime errors:
```
ERRO[0006] TypeError: Cannot read property 'lat' of undefined
running at generateBody (file:///home/aivaras/Work/traveltime-benchmarks/scripts/routes.js:85:53(12))
mainScenat file:///home/aivaras/Work/traveltime-benchmarks/scripts/routes.js:51:49(60)  executor=constant-arrival-rate scenario=mainScenario source=stacktrace
```

Reply:
in fact other endpoints seem to have this issue as well:
```
ERRO[0006] TypeError: Cannot read property 'lat' of undefined
running at generateBody (file:///home/aivaras/Work/traveltime-benchmarks/scripts/time-map.js:76:51(14))
mainScenat file:///home/aivaras/Work/traveltime-benchmarks/scripts/time-map.js:49:47(68)  executor=constant-arrival-rate scenario=mainScenario source=stacktrace
```

Reply:
how are you trying to run it?

Reply:
> how are you trying to run it?

`k6 run -e APP_ID=xxx -e API_KEY=yyy -e COUNTRY=us_estn -e RPM=10 scripts/routes.js `

Reply:
wrong country code

Reply:
for json requests it's just us